erpnext/accounts/doctype/payment_entry/payment_entry.py

```diff
@@ -350,7 +350,10 @@ def set_status(self):
 			self.status = 'Draft'
 
 	def check_reference_details(self):
-		if self.mode_of_payment == "Check" and not (self.reference_no and self.reference_date):
+		"""allow user to create Payment Entry as a draft(if mode of payment is 'Check' and Payment Type is 'Pay')
+		 and check if references are allocated before submiting the Document"""
+
+		if (self.mode_of_payment == "Check" and self.payment_type == "Pay") and not (self.reference_no and self.reference_date):
 			frappe.throw(_("Please enter Reference No and Reference Date before submitting."))
```

Comment:
- We can handle this in js itself, no need to add a validation here in py, when we can `toggle_reqd` via js

Reply:
hey @hrwX , we dont want to toggle the required but we want to allow user to at least able to save it as draft. but to make sure before submitting user enter the check details. we have this condition.